src/popover/container.tsx

```diff
@@ -47,145 +48,181 @@ export default function PopoverContainer({
   size,
   fixedWidth,
   variant,
+  keepPositionOnResize,
 }: PopoverContainerProps) {
   const bodyRef = useRef<HTMLDivElement | null>(null);
   const contentRef = useRef<HTMLDivElement | null>(null);
   const popoverRef = useRef<HTMLDivElement | null>(null);
   const arrowRef = useRef<HTMLDivElement | null>(null);
+  const internalPositionRef = useRef<InternalPosition | null>(null);
```

Comment:
This one is shaky. It works because every time the internal position is set a render as triggered by also setting the popoverStyle.

I would rather keep the state as is and introduce a new variable previousInternalPositionRef.

```
const shouldKeepPosition = keepPositionOnResize && onResize;
const fixedInternalPosition = shouldKeepPosition && previousInternalPositionRef.current;

// ...
        
if (!shouldKeepPosition) {
  setInternalPosition(newInternalPosition);
}
setPopoverStyle(...);
```

wdyt?

Reply:
Got it, done it in [e61c627](https://github.com/cloudscape-design/components/pull/1756/commits/e61c627220c339f96c89589bcf30c0ce8e3d9169).

I am calling `setInternaPosition` unconditionally though because `calculatePosition` will always return the value of the passed-in `fixedInternalPosition` as `internalPosition` if it is defined, so the permanency of this value is already guaranteed this way.